libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -528,6 +528,8 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			{
 				bool shortcutTaken = false;
 				if (auto identifier = dynamic_cast<Identifier const*>(&_functionCall.expression()))
+				{
+					solAssert(!function.bound(), "");
```

Comment:
Do you remember the reason for this?

Reply:
I can't say I remember from our conversation.
I'm thinking that since it is an `Identifier`, it can't be bound. Then, this prevents the label for the bound function call from being pushed here.

Reply:
Oh wow, I think there is a bug here.

Reply:
Ah no, I see, it's fine, but we either need the assert or change the assert into a check.
Please also add the following test, perhaps even a semantics test:
```
library L {
  function g(function() internal _t) internal pure returns (uint) { return _t(); }
}
contract C {
  using L for *;
  function f() public pure {
    t.g();
  }
  function t() public pure returns (uint)  { return 7; }
}
```